File: lctl/lctl.nim
import os, system, httpclient, json, strutils

proc change_light(light: int, state: int) =
    let client = newHttpClient()
    let body = %*{
        "light": light,
        "state": state,
    }
    let response = client.request("http://[2606:a000:1127:4082:ba27:ebff:fe59:23c]/update", httpMethod = HttpPost, body = $body)

if paramCount() != 2:
    echo "usage: lctl <light> <state>"
    quit(QuitFailure)

let params = commandLineParams()
try:
    let light = parseInt(params[0])
    let state = parseInt(params[1])

    change_light(light, state)
except ValueError:
    echo "Params must be integers"
    quit(QuitFailure)

File: lctl/lctl.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"net/http"
	"os"
	"strconv"
)

type State struct {
	Light int `json:light`
	State int `json:state`
}

func main() {
	if len(os.Args) != 3 {
		exit()
	}

	light, errLight := strconv.Atoi(os.Args[1])
	state, errState := strconv.Atoi(os.Args[2])

	if errLight != nil || errState != nil {
		exit()
	}

	sendCode(State{light, state})
}

func sendCode(state State) {
	body, _ := json.Marshal(state)
	http.Post("http://IP_ADDRESS/update",
		"application/json", bytes.NewBuffer(body))
}

func exit() {
	fmt.Println("usage: lctl <light> <state>")
	os.Exit(1)
}
